prow/plank/reconciler_test.go

```diff
@@ -328,9 +339,24 @@ func TestProwJobIndexer(t *testing.T) {
 			expected: []string{prowJobIndexKeyAll},
 		},
 		{
-			name:     "Changing name changes notCompletedByName index",
-			modify:   func(pj *prowv1.ProwJob) { pj.Spec.Job = "some-name" },
-			expected: []string{prowJobIndexKeyAll, prowJobIndexKeyPending, pendingTriggeredIndexKeyByName("some-name")},
+			name: "Changing name changes pendingTriggeredIndexKeyByName index",
+			modify: func(pj *prowv1.ProwJob) { pj.Spec.Job = "some-name" },
+			expected: []string{
+				prowJobIndexKeyAll,
+				prowJobIndexKeyPending,
+				pendingTriggeredIndexKeyByName("some-name"),
+				pendingTriggeredIndexKeyByJobQueueName(pjJobQueue),
+			},
+		},
+		{
+			name: "Changing job queue name changes pendingTriggeredIndexKeyByJobQueueName index",
+			modify: func(pj *prowv1.ProwJob) { pj.Spec.JobQueueName = "some-name" },
+			expected: []string{
+				prowJobIndexKeyAll,
+				prowJobIndexKeyPending,
+				pendingTriggeredIndexKeyByName(pjName),
+				pendingTriggeredIndexKeyByJobQueueName("some-name"),
+			},
 		},
 	}
 
```

Comment:
Can we please also add a testcase in `TestMaxConcurency` (in the prow/plank/controller_test.go file) to actually test the functionality rather than just testing the indexer change?

Reply:
Good idea, implemented. Since `canExecuteConcurrentlyPerJob` and `canExecuteConcurrentlyPerQueue` share `countPendingOrOlderTriggeredMatchingPJs` logic I added only two tests:
- Verify that limit `0` means that there is no limit
- Verify that a job can't be executed if the job queue is full